cabal-install/src/Distribution/Client/VCS.hs

```diff
@@ -467,7 +467,7 @@ vcsGit =
     vcsCloneRepo verbosity prog repo srcuri destdir =
       [programInvocation prog cloneArgs]
         -- And if there's a tag, we have to do that in a second step:
-        ++ [git (resetArgs tag) | tag <- maybeToList (srpTag repo)]
+        ++ [git (step tag) | tag <- maybeToList (srpTag repo), step <- [resetArgs]]
```

Comment:
Don't understand this change: what's the point of creating a singleton list `[resetArgs]` just to destroy it immediately?